src/main/default/classes/TriggerHandler.cls

```diff
@@ -65,51 +73,66 @@ public class TriggerHandler {
 
         if (handlers != null && !handlers.isEmpty()) {
             for (HandlerInterface h : handlers) {
-                if (!handlerDisabled(h) && !objectHandlerDisabled(sObjectType, h)) {
+                if (!handlerDisabled(h)) {
                     h.handle();
                 }
             }
         }
     }
 
-    private static Set<String> disabledObjects = new Set<String>();
-    private static Set<String> disabledHandlers = new Set<String>();
-    private static Map<String, Set<String>> disabledObjectHandlers = new Map<String, Set<String>>();
-    @TestVisible
-    private static void cacheTriggerControl(Trigger_Control__mdt[] triggerControls) {
-        disabledObjects.clear();
-        disabledHandlers.clear();
-        disabledObjectHandlers.clear();
-        for (Trigger_Control__mdt triggerControl : triggerControls) {
-            String objectName = triggerControl.Object__c == null ? null : triggerControl.Object__c.toLowerCase();
-            String handlerName = triggerControl.Class__c == null ? null : triggerControl.Class__c.toLowerCase();
-            if (handlerName == null && objectName != null) {
-                disabledObjects.add(objectName);
-            } else if (handlerName != null && objectName == null) {
-                disabledHandlers.add(handlerName);
-            } else if (handlerName != null && objectName != null) {
-                if (!disabledObjectHandlers.containsKey(objectName)) {
-                    disabledObjectHandlers.put(objectName, new Set<String>());
-                }
-                disabledObjectHandlers.get(objectName).add(handlerName);
+    private void doAutomaticBinding() {
+        if (!triggerConfig.enabled) {
+            return;
+        }
+        for (String handlerName : triggerConfig.handlerOrder) {
+            TriggerConfigService.HandlerConfig handlerConfig = triggerConfig.handlers.get(handlerName);
+            // do not bind if auto bind is off or if handler is disabled
+            if (!handlerConfig.autoBind || !handlerConfig.enabled) {
+                continue;
+            }
+            Type t = getType(handlerConfig.className);
+            if (t == null) {
+                System.debug(handlerConfig.className + '? Looks like that handler class does not exist!');
```

Comment:
nitpick, message a little soft, we know the class doesn't exist, let's just say it, also unclear what the first part is "ClassName? Looks like the handler class doesn't exist" the class name isn't really a question

why not "Handler class not found: {className}" and "Handler class missing TriggerHandler.Handler interface: {className}"

Reply:
Definitely need a better message. Didn't go back an work on this more since I'm not sure what to do with it - debugging is definitely not enough.

Platform events is a good idea. A log might be an option as well, but don't really want to add objects into the package.